OpenRA.Game/Network/Order.cs

```diff
@@ -127,17 +128,29 @@ public static Order Deserialize(World world, BinaryReader r)
 
 								case TargetType.Terrain:
 									{
-										if (world != null)
-											target = Target.FromCell(world, (CPos)r.ReadInt2());
```

Comment:
Following my comment from the serialize case: we want to keep track of the cell if it was available, which then avoids the need for the workarounds below.  It may be easiest to send all 6 components of the cell and pos across the wire.